### ISSUE TITLE
[ML] Data frame stats request logs an error if the internal index has not been created yet

### ISSUE THREAD

**Issue by dolaru** [elastic/elasticsearch#42176]
**Elasticsearch version** (`bin/elasticsearch --version`): 
Version: 7.2.0-SNAPSHOT, Build: default/tar/70ea3cf/2019-05-15T01:25:58.368763Z, JVM: 11.0.2

**Description of the problem including expected versus actual behavior**:
If a data frame has never been created, the `.data-frame-internal-1` index is not present. If you try to perform a `GET` query on the `_data_frame/transforms/_stats` API, a correct response is returned but the following error is logged in the ES log:
`[2019-05-16T08:50:05,374][ERROR][o.e.x.d.a.TransportGetDataFrameTransformsStatsAction] [dolaru-elastic-mbp.local] failed to expand ids
org.elasticsearch.index.IndexNotFoundException: no such index [.data-frame-internal-1]`

**Steps to reproduce**:
On a deployment where data frames were never created:
 1. Do a `GET _data_frame/transforms/_stats` API request
 2. Notice that the request is successful, but an `IndexNotFoundException` is logged in the ES log for the index `.data-frame-internal-1`

**Provide logs (if relevant)**:
```
[2019-05-16T08:50:05,374][ERROR][o.e.x.d.a.TransportGetDataFrameTransformsStatsAction] [dolaru-elastic-mbp.local] failed to expand ids
org.elasticsearch.index.IndexNotFoundException: no such index [.data-frame-internal-1]
	at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.indexNotFoundException(IndexNameExpressionResolver.java:761) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.innerResolve(IndexNameExpressionResolver.java:713) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.resolve(IndexNameExpressionResolver.java:669) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:163) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:138) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.search.TransportSearchAction.resolveLocalIndices(TransportSearchAction.java:456) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.search.TransportSearchAction.executeSearch(TransportSearchAction.java:470) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.search.TransportSearchAction.executeLocalSearch(TransportSearchAction.java:399) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.search.TransportSearchAction.lambda$doExecute$3(TransportSearchAction.java:211) ~[elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.index.query.Rewriteable.rewriteAndFetch(Rewriteable.java:114) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.index.query.Rewriteable.rewriteAndFetch(Rewriteable.java:87) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.search.TransportSearchAction.doExecute(TransportSearchAction.java:241) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.search.TransportSearchAction.doExecute(TransportSearchAction.java:86) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:145) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.apply(SecurityActionFilter.java:123) [x-pack-security-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:143) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:121) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:64) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.client.node.NodeClient.executeLocally(NodeClient.java:83) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.client.node.NodeClient.doExecute(NodeClient.java:72) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:394) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.client.support.AbstractClient.search(AbstractClient.java:527) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.core.ClientHelper.executeAsyncWithOrigin(ClientHelper.java:74) [x-pack-core-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.dataframe.persistence.DataFrameTransformsConfigManager.expandTransformIds(DataFrameTransformsConfigManager.java:213) [data-frame-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.dataframe.action.TransportGetDataFrameTransformsStatsAction.doExecute(TransportGetDataFrameTransformsStatsAction.java:124) [data-frame-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.dataframe.action.TransportGetDataFrameTransformsStatsAction.doExecute(TransportGetDataFrameTransformsStatsAction.java:64) [data-frame-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:145) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.apply(SecurityActionFilter.java:123) [x-pack-security-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:143) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:121) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:64) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.client.node.NodeClient.executeLocally(NodeClient.java:83) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.client.node.NodeClient.doExecute(NodeClient.java:72) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:394) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.dataframe.rest.action.RestGetDataFrameTransformsStatsAction.lambda$prepareRequest$0(RestGetDataFrameTransformsStatsAction.java:35) [data-frame-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.rest.BaseRestHandler.handleRequest(BaseRestHandler.java:113) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter.handleRequest(SecurityRestFilter.java:69) [x-pack-security-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:240) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.rest.RestController.tryAllHandlers(RestController.java:337) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:174) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.http.AbstractHttpServerTransport.dispatchRequest(AbstractHttpServerTransport.java:320) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.http.AbstractHttpServerTransport.handleIncomingRequest(AbstractHttpServerTransport.java:370) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.http.AbstractHttpServerTransport.incomingRequest(AbstractHttpServerTransport.java:299) [elasticsearch-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.http.netty4.Netty4HttpRequestHandler.channelRead0(Netty4HttpRequestHandler.java:66) [transport-netty4-client-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at org.elasticsearch.http.netty4.Netty4HttpRequestHandler.channelRead0(Netty4HttpRequestHandler.java:31) [transport-netty4-client-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(Netty4HttpPipeliningHandler.java:58) [transport-netty4-client-7.2.0-SNAPSHOT.jar:7.2.0-SNAPSHOT]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) [netty-codec-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) [netty-codec-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) [netty-codec-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) [netty-codec-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) [netty-codec-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297) [netty-codec-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:287) [netty-handler-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:682) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:582) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:536) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [netty-transport-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:906) [netty-common-4.1.35.Final.jar:4.1.35.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.35.Final.jar:4.1.35.Final]
	at java.lang.Thread.run(Thread.java:835) [?:?]
```

**Comment by hendrikmuhs** [comment]
@davidkyle fixed this issue in #41752

see:
https://github.com/elastic/elasticsearch/commit/9d94d575314289e88067bee4d2b88ff0dc05af5a#diff-d632b8918e949afa576a0c93b5dcd241L135

**Comment by dolaru** [comment]
I can confirm that fixed it. Thanks!
